### PR TITLE
VMware: Raw Device Mapping

### PR DESCRIPTION
**Pull request by sandrich**
##### SUMMARY
This is work in progress for raw device mapping which should tackle #24837

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
vmware_guest_disk

##### ADDITIONAL INFORMATION
The enhancement allows to map raw devices into the VM. The current implementation is working but is missing some tests and code cleanup. Also currently only physicalMode compatibility mode is supported.

Also thanks to @radudd for his contributions

```paste below

```

### CODE REVIEW

**Comment by sandrich** [comment]
I was wondering how to write tests for raw device mapping. The test esx environment would require some additional disks I could use. Any help would be appreciated.

**Comment by sandrich** [comment]
ready_for_review

File: lib/ansible/modules/cloud/vmware/vmware_guest_disk.py [hunk 1]
```diff
@@ -572,7 +605,7 @@ def sanitize_disk_inputs(self):
 
             # Type of Disk
             disk_type = disk.get('type', 'thick').lower()
-            if disk_type not in ['thin', 'thick', 'eagerzeroedthick']:
+            if disk_type not in ['thin', 'thick', 'eagerzeroedthick', 'raw']:
                 self.module.fail_json(msg="Invalid 'disk_type' specified for disk index [%s]. Please specify"
                                           " 'disk_type' value from ['thin', 'thick', 'eagerzeroedthick']." % disk_index)
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_guest_disk.py]:
```suggestion
                                          " 'disk_type' value from ['thin', 'thick', 'eagerzeroedthick', 'raw']." % disk_index)
```

**Review by Akasurde — CHANGES_REQUESTED** [review]
Could you please rebase this PR? Thanks for the contribution.

**Comment by sandrich** [comment]
Screwed this one up. Will recreate

**Pull request closed without merging**